src/Microsoft.ML.Core/Utilities/Utils.cs

```diff
@@ -615,37 +615,75 @@ public static string ExtractLettersAndNumbers(string value)
             return Regex.Replace(value, "[^A-Za-z0-9]", "");
         }
 
-        public static bool IsSorted(IList<float> values)
+        /// <summary>
+        /// Checks that an input IList is monotonically increasing.
+        /// </summary>
+        /// <param name="values">An array of values</param>
+        /// <returns>True if the array is monotonically increasing; false otherwise.
+        /// ILists containing NaN values are considered to be not monotonically increasing.</returns>
+        public static bool IsMonotonicallyIncreasing(IList<float> values)
```

Comment:
>IsMonotonicallyIncreasing [](start = 27, length = 25)

```
        public static bool IsMonotonicallyIncreasing<T>(IList<T> values) where T : IComparable
        {
            if (values.Count <= 1)
                return true;

            var previousValue = values[0];
            var listLength = values.Count;
            for (int i = 1; i < listLength; i++)
            {
                var currentValue = values[i];
                // Inverted check for NaNs
                if (currentValue.CompareTo(previousValue) < 0)
                    return false;

                previousValue = currentValue;
            }

            return true;
        }
```
Hope I get CompareTo condition right.
That way you don't have to define second function
also I'm not sure regarding first if, does empty array is actually MonotonicallyIncreasing?

Reply:
regarding 1.
Array is already IList, you don't need to have two implementations.
https://docs.microsoft.com/en-us/dotnet/api/system.array?view=netframework-4.7.2

2. unfortunately, you are right.
But you still can have only IList implementation!

Reply:
You are totally right.

And for the original question, as to whether an empty array is monotonically increasing, I'm keeping the original implementation so we don't cause any downstream effects. I guess you could say that it's not not monotonically increasing....

Reply:
Updating s.t. all contracts specify `IList` as that is a broader level of support, as you point out.